ui/desktop/src/App.tsx

```diff
@@ -4,23 +4,59 @@ import { useNavigate } from 'react-router-dom';
 import LauncherWindow from './LauncherWindow';
 import ChatWindow from './ChatWindow';
 import ErrorScreen from './components/ErrorScreen';
+import { ConfirmationModal } from './components/ui/ConfirmationModal';
 import 'react-toastify/dist/ReactToastify.css';
 import { ToastContainer } from 'react-toastify';
 import { ModelProvider } from './components/settings/models/ModelContext';
 import { ActiveKeysProvider } from './components/settings/api_keys/ActiveKeysContext';
 
 export default function App() {
   const [fatalError, setFatalError] = useState<string | null>(null);
+  const [modalVisible, setModalVisible] = useState(false);
```

Comment:
nit: can we use a more descriptive name of which modal? Something like `confirmExtensionModalVisible`